velox/type/Type.h

```diff
@@ -65,6 +67,7 @@ enum class TypeKind : int8_t {
   VARCHAR = 7,
   VARBINARY = 8,
   TIMESTAMP = 9,
+  DATE = 17,
```

Comment:
It it strange to have 17 in between 9 and 10. Let's move this enum to maintain intuitive order. Alternatively, I think it is OK to change the values, so you can keep DATE here, but change the value to 10 and change all the following values accordingly. CC: @zzhao0